src/common/components/formikPersist/__tests__/FormikPersist.test.tsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable max-len */
+import { render, waitFor } from '@testing-library/react';
+import { Formik, FormikProps } from 'formik';
+import * as React from 'react';
+
+import Persist from '../FormikPersist';
+
+beforeEach(() => {
+  // values stored in tests will also be available in other tests unless you run
+  localStorage.clear();
+  sessionStorage.clear();
+});
+
+const formName = 'form-name';
+
+test('attempts to rehydrate on mount', async () => {
+  let injected: FormikProps<{ name: string }>;
+
+  (localStorage.getItem as jest.Mock).mockReturnValueOnce(
+    '{"values":{"name":"Name from local storage"},"errors":{},"touched":{},"isSubmitting":false,"isValidating":false,"submitCount":0,"initialValues":{"name":"Test name"},"initialErrors":{},"initialTouched":{},"isValid":true,"dirty":true,"validateOnBlur":true,"validateOnChange":true,"validateOnMount":false}'
+  );
+
+  render(
+    <Formik initialValues={{ name: 'Test name' }} onSubmit={jest.fn()}>
+      {(props: FormikProps<{ name: string }>) => {
+        injected = props;
+
+        return (
+          <div>
+            <Persist name={formName} debounceTime={0} />
+          </div>
+        );
+      }}
+    </Formik>
+  );
+
+  expect(localStorage.getItem).toHaveBeenCalled();
+
+  expect(injected.values.name).toEqual('Name from local storage');
+
+  await waitFor(() => {
+    injected.setValues({ name: 'changed value' });
+  });
```

Comment:
seems strange way to use waitFor. Why is it used here? Is injected.setValues undefined at first?

Reply:
The idea was to get rid of "An update to Formik inside a test was not wrapped in act" error. Changes waitFor to act

Reply:
great, actually I think act is the correct way to do this :)